Gordon360/Services/HousingService.cs

```diff
@@ -531,4 +557,228 @@ public bool ChangeApplicationDateSubmitted(int applicationID)
+    {
+        var preferences = preferenceList.Select((preference) => new Preference { ApplicationID = application_id, Preference1 = preference });
+        context.Preference.AddRange(preferences);
+        context.SaveChanges();
+    }
+
+    /// <summary>
+    /// update the information of due date
+    /// </summary>
+    /// <param name="dueDate"> The due date of the application </param>
+    public async Task UpdateDueDateAsync(string dueDate)
+    {
+        // This code here is not working correctly. It cannot save the changes.
+        var myDueDate = context.Config.FirstOrDefault(d => d.Key == "housing_lottery_due_date");
+        myDueDate.Value = dueDate;
+        await context.SaveChangesAsync();
+    }
+
+    /// <summary>
+    /// Remove the particular user from the current application by setting her/his active flag to 0
+    /// </summary>
+    /// <returns> Whether or not this was successful </returns>
+    public bool RemoveUser(string username)
+    {
+        var email = context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.email;
+        var existActiveApplicant = context.Applicant.FirstOrDefault(x => (x.Applicant1 == email) && (x.Active == 1));
+        if (existActiveApplicant == null)
+        {
+            throw new ResourceNotFoundException() { ExceptionMessage = "The applicant could not be found" };
+        }
+        existActiveApplicant.Active = 0;
+        context.SaveChanges();
+        return true;
+    }
+
+    /// <summary>
+    /// Gets an array of preferences
+    /// </summary>
+    /// <returns> An array of preferences </returns>
+    public IEnumerable<HousingPreferenceViewModel> GetAllPreferences()
+    {
+        var activeApplicants = context.Applicant.Where(a => a.Active == 1);
+
+        var result = context.Preference
+        .Join(activeApplicants, p => p.ApplicationID, a => a.ApplicationID, (p, a) => new HousingPreferenceViewModel
+        {
+            ApplicationID = p.ApplicationID,
+            Preference1 = p.Preference1,
+        }).Distinct();
+        if (result == null || !result.Any())
+        {
+            throw new ResourceNotFoundException() { ExceptionMessage = "The preferences could not be found." };
+        }
+        return result.ToArray();
+    }
+
+    /// <summary>
+    /// Gets an array of preferences of this user
+    /// </summary>
+    /// <returns> An array of preferences </returns>
+    public IEnumerable<HousingPreferenceViewModel> GetUserPreference(string username)
+    {
+        var email = context.ACCOUNT.FirstOrDefault(a => a.AD_Username == username)?.email;
+        var applicationID = context.Applicant.FirstOrDefault(a => (a.Applicant1 == email) && (a.Active == 1))?.ApplicationID;
+        var preference = context.Preference.Where(a => a.ApplicationID == applicationID).Select(ph => (HousingPreferenceViewModel)ph);
+        return preference.ToArray();
+    }
```

Comment:
As I noted in the controller, preferences are technically not scoped to a user, so I would recommend taking in an application ID and using that. It makes the method simpler and more reusable (modular).